esp-hal/src/hmac.rs

```diff
@@ -256,12 +254,12 @@ impl<'d> Hmac<'d> {
             }
         }
 
-        Ok(remaining)
+        Some(remaining)
     }
 
-    fn flush_data(&mut self) -> nb::Result<(), Infallible> {
+    fn flush_data(&mut self) -> Option<()> {
```

Comment:
This applies to other drivers as well, but `Option<()>` as a return type feels a bit wrong to me. Perhaps we should just block instead, given that these are blocking APIs to begin with?

Reply:
I'd say it's better to return Boolean rather than block. Gives users more options/flexibility